src/main/resources/assets/js/components/AppModalComponent.jsx

```diff
@@ -31,17 +33,49 @@ var AppModalComponent = React.createClass({
     rollBackApp: React.PropTypes.func.isRequired,
     scaleApp: React.PropTypes.func.isRequired,
     suspendApp: React.PropTypes.func.isRequired,
-    tasksFetchState: React.PropTypes.number.isRequired
+    tasksFetchState: React.PropTypes.number.isRequired,
+    router: React.PropTypes.object.isRequired
   },
 
   getInitialState: function () {
+    var appid = this.props.model.get("id");
+    var activeTabId;
+
+    var tabs = _.reduce(tabsTemplate, function (current, tab) {
```

Comment:
I'm okay with it being a separate task as long as this doesn't get merged until that happens. Having routing at this level has way too many repercussions and will be difficult to remove in the future.